site/src/pages/RoadmapPage/YearModal.tsx

```diff
@@ -0,0 +1,154 @@
+import React, { FC, useState } from 'react';
+import { Button, Form, Modal } from 'react-bootstrap';
+import { PlannerYearData } from '../../types/types';
+import './YearModal.scss';
+
+interface YearPopupQuarter {
+  id: string;
+  name: string;
+  checked?: boolean;
+}
+
+interface CourseYearModalProps {
+  placeholderName: string;
+  placeholderYear: number;
+  show: boolean;
+  setShow: React.Dispatch<React.SetStateAction<boolean>>;
+  type: 'add' | 'edit';
+  saveHandler: (x: PlannerYearData) => void;
+  currentQuarters: string[];
+}
+
+const CourseYearModal: FC<CourseYearModalProps> = (props) => {
+  const { placeholderName, placeholderYear, show, setShow, type, saveHandler, currentQuarters } = props;
+  const [validated, setValidated] = useState(false);
+
+  const [name, setName] = useState(placeholderName);
+  const [year, setYear] = useState(placeholderYear);
+
+  const quarterValues: (selectedNames: string[]) => YearPopupQuarter[] = (data: string[]) => {
+    const base: YearPopupQuarter[] = [
+      { id: 'fall', name: 'Fall' },
+      { id: 'winter', name: 'Winter' },
+      { id: 'spring', name: 'Spring' },
+      { id: 'summer 1', name: 'Summer 1' },
+      { id: 'summer 2', name: 'Summer 2' },
+      { id: 'summer 10 Week', name: 'Summer 10 Week' },
+    ];
+    data.forEach((name) => {
+      const match = base.find((q) => q.id === name);
+      if (match) match.checked = true;
+    });
+    return base;
+  };
+
+  const [quarters, setQuarters] = useState<YearPopupQuarter[]>(quarterValues(currentQuarters));
+  const quarterCheckboxes = quarters.map((q, i) => {
+    const handleClick = (i: number) => {
+      const newQuarters = quarters.slice();
+      newQuarters[i].checked = !newQuarters[i].checked;
+      setQuarters(newQuarters);
+    };
+    return (
+      <Form.Check
+        key={q.id}
+        type="checkbox"
+        id={'quarter-checkbox-' + q.id}
+        label={q.name}
+        value={q.id}
+        checked={q.checked ?? false}
```

Comment:
Nullish coalescing with false shouldn't be necessary.
```suggestion
        checked={q.checked}
```

Reply:
This is necessary. Without this, React will give a warning that you're now assigning a new prop instead of just changing its value.

Reply:
This is because undefined is equivalent to not defining the prop, in which case I think adding a comment about this would be good so it's not changed unintentionally.
something like `// must be assigned a value that is not undefiend`